ui/app/resources/base-factory.ts

```diff
@@ -3,18 +3,27 @@
  * SPDX-License-Identifier: BUSL-1.1
  */
 
+import { getOwner, setOwner } from '@ember/owner';
+
+import type Owner from '@ember/owner';
+
 abstract class BaseResource<T> {
   // pass data that the resource should represent (typically from an API response) to constructor
   // object properties will be assigned to class instance
   // extending classes can define getters and additional properties/methods that are required widely across the app
-  constructor(readonly data: T) {
+  constructor(data: T, context?: unknown) {
     Object.assign(this, data) as T;
+    // pass in context (this) of Ember class (route, component etc.) where the resource is being constructed
+    // this will be used to set the owner on the class so that services can be injected (if required)
+    if (context) {
+      setOwner(this, getOwner(context) as Owner);
+    }
   }
```

Comment:
The comment sums it up but basically since this is a native js class that is not instantiated by Ember there is no owner set. Ember will throw an error when attempting to inject a service on a class without an owner. Since resources are typically used within Ember based classes (route, controller, component) we can pass in the context (this) to the constructor and get the owner from it and then set that owner on the resource class so that service injection can happen.